internal/cmd/branch/promote.go

```diff
@@ -78,13 +79,46 @@ func PromoteCmd(ch *cmdutil.Helper) *cobra.Command {
 				}
 			}
 
+			getReq := &ps.GetPromotionRequestRequest{
+				Organization: ch.Config.Organization,
+				Database:     source,
+				Branch:       branch,
+			}
+
+			for promotionRequest.State == "pending" {
+				promotionRequest, err = client.DatabaseBranches.GetPromotionRequest(cmd.Context(), getReq)
+				if err != nil {
+					switch cmdutil.ErrCode(err) {
+					case ps.ErrNotFound:
+						return fmt.Errorf("promotion request for branch %s does not exist in database %s", printer.BoldBlue(branch), printer.BoldBlue(source))
+					default:
+						return cmdutil.HandleError(err)
+					}
+				}
+
+				time.Sleep(1 * time.Second)
```

Comment:
We should have a timeout, like 4 or 5 minutes and bailout. Otherwise, in case of a wrong API, this will loop forever. For that, I would refactor this for loop into a function to make it easy, and use a `for/select` idiom with two tickers. On being a one-second ticker that checks the state, and the other select channel will be `time.After(timeout)` and return an error.  Here is an example:

```go
func waitPromoteState() (bool, error) {
    timeout := time.After(5* time.Minute)
    ticker := time.NewTicker(time.Second)
    defer ticker.Stop()


    for {
        select {
        case <-timeout:
            return false, errors.New("timed out")
        case <-ticker.C:
            // check promote status 
        }
}
```